target_csv.py

```diff
@@ -146,11 +134,11 @@ def main():
                     'the config parameter "disable_collection" to true')
         threading.Thread(target=send_usage_stats).start()
 
-    input = io.TextIOWrapper(sys.stdin.buffer, encoding='utf-8')
-    state = persist_lines(config.get('delimiter', ','),
-                          config.get('quotechar', '"'),
-                          input)
-        
+    input_messages = io.TextIOWrapper(sys.stdin.buffer, encoding='utf-8')
+    state = persist_messages(config.get('delimiter', ','),
+                             config.get('quotechar', '"'),
+                             input_messages)
```

Comment:
Actually if you install target-csv before a new tap, like tap-exchangeratesapi, it crashes, because target-csv now requires a specific outdated version of singer-python to work. I changed the code in setup.py to fix the problem. What do you think?

Reply:
Ah. This comes up fairly regularly and I'm not sure exactly how to make it clearer. As is outlined in the [Running and Developing guide](https://github.com/singer-io/getting-started/blob/master/docs/RUNNING_AND_DEVELOPING.md#running-singer-with-python), the intended way to run taps and targets is in separate virtual environments. This avoids issues exactly like you're describing. Our [official stance on loose version requirements](https://github.com/singer-io/singer-python/issues/80#issuecomment-432814139) (as official as a comment on an issue can get!) shows our Java roots but we think ultimately yields a more sane development experience. :)

So my recommendation would be to reverse the setup.py change and maybe link to that getting started guide?

Reply:
Ok, done! I think the development experience may be a bit more sane, but for newcomers to the library is very difficult to use. I mean, a lot of python users don't even know how to use a single virtualenv, imagine using two, one for each library. The end user just suffer more than it is necessary in my opinion. Just food for thought.

Do you think it is necessary to show a gist?